include/utils.hxx

```diff
@@ -69,19 +69,46 @@ void free_cmatrix(dcomplex** cm);
 
 BoutReal randomu(); // Get Random number between 0 and 1
 
-BoutReal SQ(BoutReal x);
-int ROUND(BoutReal x);
-int BOUTMAX(int a, int b);
-BoutReal BOUTMAX(BoutReal a, BoutReal b);
-BoutReal BOUTMAX(BoutReal a, BoutReal b, BoutReal c);
-BoutReal BOUTMIN(BoutReal a, BoutReal b);
-BoutReal BOUTMIN(BoutReal a, BoutReal b, BoutReal c);
-bool is_pow2(int x); // Check if a number is a power of 2
-BoutReal SIGN(BoutReal a); // Return +1 or -1 (0 -> +1)
+template <typename T>
+T SQ(T t){
+  return t*t;
+}
+
+inline int ROUND(BoutReal x){
+  return (x > 0.0) ? (int) (x + 0.5) : (int) (x - 0.5);
+}
+
+template <typename T>
+T BOUTMAX(T a){
+  return a;
+}
+template <typename T, typename... Args>
+T BOUTMAX(T a,T b,Args... args){
+  T c = BOUTMAX(b,args...);
+  return c > a? c :a;
```

Comment:
Incredibly minor, but the whitespace here (and L98) is inconsistent, this would look better as:

      return c > a ? c : a;